plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ServerAdapterHandler.java

```diff
@@ -75,31 +72,34 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 	 * Finds the OpenShift server corresponding to the selection or prompts the
 	 * user to create one.
 	 * 
-	 * @param selectedResource
+	 * @param selectedResourceModel
 	 *            the selected OpenShift {@link IResourceUIModel}
 	 * 
 	 * @return the matching OpenShift {@link IServer} or <code>null</code> if
 	 *         none was found or user cancelled the creation operation.
 	 */
-	private IServer getOpenShiftServer(final IResourceUIModel selectedResource) {
-		if (selectedResource != null) {
-			if (selectedResource.getResource() instanceof IService) {
-				final IService selectedService = (IService) selectedResource.getResource();
-				final Connection connection = ConnectionsRegistryUtil.safeGetConnectionFor(selectedService);
-				return openOrCreateServerAdapter(selectedService, connection);
-			} else if (selectedResource.getResource() instanceof IRoute) {
-				final IRoute selectedRoute = (IRoute) selectedResource.getResource();
+	private IServer getOpenShiftServer(final IResourceUIModel selectedResourceModel) {
```

Comment:
yes that's the alternative to this. Either like done here or get the resources from deployment resource model. The 1st problem with the resource model is  https://issues.jboss.org/browse/JBIDE-21326 . The other is that it's loading far too much, causing https://issues.jboss.org/browse/JBIDE-21858. I thus took this other approach using what we already have.